src/Controller/ProjectController.php

```diff
@@ -32,7 +32,8 @@ public function create(Request $request, ProjectRepositoryInterface $repository)
     {
         //TODO: validation
         $data = json_decode($request->getContent(), true);
-        $project = new Project(ProjectId::new(), $data['name'], $data['description']);
+        $id = ProjectId::fromString($data['id']);
```

Comment:
Beside the user input validation what's done in the TODO part :), in the ValueObject ProjectId from string we should add check for the pattern and throw an InvalidIdException (what is not implemented yet). ValueObject must be always valid.